src/sentry/spans/buffer/redis.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+
+from django.conf import settings
+from sentry_redis_tools.clients import RedisCluster, StrictRedis
+
+from sentry.utils import redis
+
+SEGMENT_TTL = 5 * 60  # 5 min TTL in seconds
+
+
+def get_redis_client() -> RedisCluster | StrictRedis:
+    return redis.redis_clusters.get(settings.SENTRY_SPAN_BUFFER_CLUSTER)
+
+
+def get_segment_key(project_id: str | int, segment_id: str) -> str:
+    return f"segment:{segment_id}:{project_id}:process-segment"
+
+
+class RedisSpansBuffer:
+    def __init__(self):
+        self.client: RedisCluster | StrictRedis = get_redis_client()
+
+    def read_segment(self, project_id: str | int, segment_id: str) -> list[str | bytes]:
+        key = get_segment_key(project_id, segment_id)
+
+        return self.client.lrange(key, 0, -1) or []
+
+    def write_span(self, project_id: str | int, segment_id: str, span: bytes) -> bool:
+        key = get_segment_key(project_id, segment_id)
+        length = self.client.rpush(key, span)
+        new_key = length == 1
+
+        if new_key:
+            self.client.expire(key, SEGMENT_TTL)
```

Comment:
No need to test the value of the key. Do a pipeline call (to send all commands at once) and use the `NX` argument to only set the expiration if it has not been set before. Like this, when you create the key, it'll set the expiration and will be no-op in subsequent calls, and you don't have to double the number of calls to Redis,

Reply:
Good suggestion! The `nx` arg on expire is only supported starting Redis version 7.0.0 ([docs](https://redis.io/commands/expire/)) and without it, the pipelining won't work since we'll need the response from the push command. 
We're on 
```
docker exec -it sentry_redis redis-cli INFO server
# Server
redis_version:5.0.14
```

So I'll leave this as is for now.